torch/quantization/fuse_modules.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import torch
+import copy
+
+import torch.nn._intrinsic.modules.fused as torch_fused
+
+def fuse_conv_bn(conv, bn):
+    r"""Given the conv and bn modules, fuses them and returns the fused module
+
+    Args:
+        conv: Module instance of type conv2d
+        bn: Spatial BN instance that needs to be fused with the conv
+
+    Examples::
+
+        >>> m1 = nn.Conv2d(10, 20, 3)
+        >>> b1 = nn.BatchNorm2d(20)
+        >>> m2 = fuse_conv_bn(m1, b1)
+    """
+    assert(conv.training == bn.training),\
+        "Conv and BN both must be in the same mode (train or eval)."
+
+    if conv.training:
+        fused_conv = torch.nn._intrinsic.modules.ConvBn2d(conv, bn)
+    else:
+        fused_conv = copy.deepcopy(conv)
+
+        w_conv = fused_conv.weight
+        b_conv = fused_conv.bias
+
+        bn_mean = bn.running_mean
+        bn_var_sqrt = torch.sqrt(bn.running_var + bn.eps)
+
+        bn_weight = bn.weight
+        bn_bias = bn.bias
+
+        if b_conv is None:
+            b_conv = bn_mean.new_zeros(bn_mean.shape)
+
+        w_conv = w_conv * (bn_weight / bn_var_sqrt).reshape([-1, 1, 1, 1])
```

Comment:
why are you reshaping it here?

Reply:
Because the w_conv and bn_weight have different dimensions: [*, 1, 1, 1] vs. [*,]. I couldn't do the broadcasting :/

Reply:
Ah, sorry, I misread what precisely you're reshaping. It makes sense (unsqueeze would be another option, but this works too)